internal/distro/rhel9/imagetype.go

```diff
@@ -327,6 +390,65 @@ func (t *imageType) Manifest(customizations *blueprint.Customizations,
 	)
 }
 
+func (t *imageType) ManifestNew(customizations *blueprint.Customizations,
+	options distro.ImageOptions,
+	repos []rpmmd.RepoConfig,
+	packageSets map[string][]rpmmd.PackageSpec,
+	containers []container.Spec,
+	seed int64) (distro.Manifest, error) {
+
+	bp := &blueprint.Blueprint{Name: "empty blueprint"}
+	err := bp.Initialize()
+	if err != nil {
+		panic("could not initialize empty blueprint: " + err.Error())
+	}
+	bp.Customizations = customizations
+
+	manifest, err := t.initializeManifest(bp, options, repos, nil, containers, seed)
+	if err != nil {
+		return distro.Manifest{}, err
+	}
+
+	return manifest.Serialize(packageSets)
+}
+
+func (t *imageType) PackageSetsNew(bp blueprint.Blueprint, options distro.ImageOptions, repos []rpmmd.RepoConfig) map[string][]rpmmd.PackageSet {
+	// merge package sets that appear in the image type with the package sets
+	// of the same name from the distro and arch
+	packageSets := make(map[string]rpmmd.PackageSet)
+
+	for name, getter := range t.packageSets {
+		packageSets[name] = getter(t)
+	}
+
+	// amend with repository information
+	globalRepos := make([]rpmmd.RepoConfig, 0)
+	for _, repo := range repos {
+		if len(repo.PackageSets) > 0 {
+			// only apply the repo to the listed package sets
+			for _, psName := range repo.PackageSets {
+				ps := packageSets[psName]
+				ps.Repositories = append(ps.Repositories, repo)
+				packageSets[psName] = ps
+			}
+		} else {
+			// no package sets were listed, so apply the repo
+			// to all package sets
+			globalRepos = append(globalRepos, repo)
+		}
+	}
+
+	// create a manifest object and instantiate it with the computed packageSetChains
+	manifest, err := t.initializeManifest(&bp, options, globalRepos, packageSets, nil, 0)
+	if err != nil {
+		// TODO: handle manifest initialization errors more gracefully, we
+		// refuse to initialize manifests with invalid config.
+		return nil
```

Comment:
you should at least print the error here (as done in #3132)

Reply:
Done!